HorizonEngine/src/HorizonEngine/GraphEditor/GraphEditor.h

```diff
@@ -0,0 +1,122 @@
+#pragma once
+#include <pch.h>
+#include <stdint.h>
+#include "imgui.h"
+#include "imgui_internal.h"
+
+namespace GraphEditor {
+
+    typedef size_t NodeIndex;
+    typedef size_t SlotIndex;
+    typedef size_t LinkIndex;
+    typedef size_t TemplateIndex;
+
+    // Force the view to be respositionned and zoom to fit nodes with Show function.
+    // Parameter value will be changed to Fit_None by the function.
+    enum FitOnScreen
+    {
+        Fit_None,
+        Fit_AllNodes,
+        Fit_SelectedNodes
+    };
```

Comment:
Ah this is cool, so it will adjust to show all the nodes?

Reply:
Yeah, exactly can fit highlight nodes and then all nodes

Reply:
sweet, look forward to having a mess around with it myself!